pkg/manifest/jsonmanifest/manifest.go

```diff
@@ -6,39 +6,27 @@ package jsonmanifest
 
 import (
 	"encoding/json"
-	"net/http"
 
 	"github.com/ethersphere/bee/pkg/manifest"
-	"github.com/ethersphere/bee/pkg/swarm"
 )
 
-var _ manifest.Parser = (*JSONParser)(nil)
-
-type JSONParser struct{}
-
-func NewParser() *JSONParser {
-	return &JSONParser{}
-}
-
-func (m *JSONParser) Parse(bytes []byte) (manifest.Interface, error) {
-	mi := &JSONManifest{}
-	err := json.Unmarshal(bytes, mi)
-
-	return mi, err
-}
-
+// verify JSONManifest implements manifest.Interface
 var _ manifest.Interface = (*JSONManifest)(nil)
 
+// JSONManifest is a JSON representation of a manifest
+// it stores manifest entries in a map based on string keys
 type JSONManifest struct {
 	Entries map[string]JSONEntry `json:"entries,omitempty"`
 }
 
+// NewManifest creates a new JSONManifest struct and returns a pointer to it
 func NewManifest() *JSONManifest {
```

Comment:
this should ideally return `manifest.Interface`, although, i think that due to the fact you're accessing the exposed `Entries` from outside the package its a no-go (at least until that's sorted). Also, once the constructor returns the interface type, `JSONManifest` can be safely unexported.

Reply:
right, i'm currently working on this through #480

Reply:
Since Entries are required to be accessed outside, I see that as a miss in the manifest.Interface interface. If manifest package defines required interfaces, they should be complete. If there is a need to go from interface to a concrete type to perform some functionality, then the interface is not complete.

I am suggesting to unexport JSONManifest and JSONEntry, use appropriate manifest interfaces instead and see where the abstraction needs to be improved.

Also have in mind that a compatible encoding is also important for different implementations in this case.

Reply:
yes, i apologize for this :pray: 

i always try to do 1 PR per issue, but stuff kept coming up through reviews with this one. i tried to separate as much as possible but clearly i could have done better :sweat_smile:  

i thought maybe having this entry business in a 2nd PR (which builds on this one) would be good to tackle maybe the serialization first, and then the entries stuff later; to try and keep things as lean as possible.

in any case, i apologize for the confusion.